File: app/src/main/java/com/musicplayer/SocyMusic/ui/album/AlbumSongAdapter.java
package com.musicplayer.SocyMusic.ui.album;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.musicplayer.SocyMusic.data.Song;
import com.musicplayer.musicplayer.R;

import java.util.List;

public class AlbumSongAdapter extends RecyclerView.Adapter<AlbumSongHolder> {
    private Context context;
    private List<Song> albumSongs;
    private ItemClickListener clickListener;

    public AlbumSongAdapter(Context context, List<Song> albumSongs) {
        this.context = context;
        this.albumSongs = albumSongs;
    }

    @NonNull
    @Override
    public AlbumSongHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        LayoutInflater inflater = (LayoutInflater) context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
        View itemView = inflater.inflate(R.layout.list_item_album_song, parent, false);
        return new AlbumSongHolder(itemView, clickListener);
    }

    @Override
    public void onBindViewHolder(@NonNull AlbumSongHolder holder, int position) {
        holder.bind(albumSongs.get(position));
    }

    @Override
    public int getItemCount() {
        return albumSongs.size();
    }

    public void setOnItemClickListener(ItemClickListener itemClickListener) {
        this.clickListener = itemClickListener;
    }

    interface ItemClickListener {

        void onItemClick(int position, View view);

        boolean onItemLongClick(int position, View view);

    }
}

File: app/src/main/kotlin/com/musicplayer/openmusic/ui/album/AlbumSongAdapter.kt
package com.musicplayer.openmusic.ui.album

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.musicplayer.openmusic.R
import com.musicplayer.openmusic.data.Song

class AlbumSongAdapter(private val context: Context, private val albumSongs: List<Song>) :
    RecyclerView.Adapter<AlbumSongHolder>() {
    private var clickListener: ItemClickListener? = null
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): AlbumSongHolder {
        val inflater = context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
        val itemView = inflater.inflate(R.layout.list_item_album_song, parent, false)
        return AlbumSongHolder(itemView, clickListener!!)
    }

    override fun onBindViewHolder(holder: AlbumSongHolder, position: Int) {
        holder.bind(albumSongs[position])
    }

    override fun getItemCount(): Int {
        return albumSongs.size
    }

    fun setOnItemClickListener(itemClickListener: ItemClickListener) {
        clickListener = itemClickListener
    }

    interface ItemClickListener {
        fun onItemClick(position: Int, view: View)
        fun onItemLongClick(position: Int, view: View): Boolean
    }
}